pyanaconda/modules/storage/storage.py

```diff
@@ -89,15 +83,31 @@ def __init__(self):
             self._zfcp_module = ZFCPModule()
             self._add_module(self._zfcp_module)
 
-        # Connect signals.
-        self.storage_changed.connect(self._auto_part_module.on_storage_reset)
-        self.storage_changed.connect(self._manual_part_module.on_storage_reset)
-        self.storage_changed.connect(self._custom_part_module.on_storage_reset)
+        # Initialize the partitioning modules.
+        self._partitioning_modules = {}
+
+        self._auto_part_module = AutoPartitioningModule()
+        self._add_partitioning_module(AUTO_PARTITIONING.object_path, self._auto_part_module)
+
+        self._manual_part_module = ManualPartitioningModule()
+        self._add_partitioning_module(MANUAL_PARTITIONING.object_path, self._manual_part_module)
+
+        self._custom_part_module = CustomPartitioningModule()
+        self._add_partitioning_module(CUSTOM_PARTITIONING.object_path, self._custom_part_module)
 
     def _add_module(self, storage_module):
         """Add a base kickstart module."""
         self._modules.append(storage_module)
 
+    def _add_partitioning_module(self, object_path, partitioning_module):
+        """Add a partitioning module."""
+        # Add the module.
+        self._modules.append(partitioning_module)
+        self._partitioning_modules[object_path] = partitioning_module
```

Comment:
It could be interesting to check if the `object_path` is already in the dict and raise an exception if it is. It couldn't happen but during all the refactorization it could happen.